app/javascript/components/character-count.js

```diff
@@ -0,0 +1,187 @@
+function CharacterCount ($module) {
+  this.$module = $module
+}
+
+CharacterCount.prototype.defaults = {
+  characterCountAttribute: 'data-maxlength',
+  wordCountAttribute: 'data-maxwords'
+}
+
+// Initialize component
+CharacterCount.prototype.init = function (options) {
+  // Check for module
+  var $module = this.$module
+  if (!$module) {
+    return
+  }
+
+  // Read options set using dataset ('data-' values) and merge them with options
+  var dataset = this.getDataset($module)
+  this.options = this.merge(options || {}, dataset)
+
+  // Determine the limit attribute (characters or words)
+  var countAttribute = (this.options.maxwords) ? this.defaults.wordCountAttribute : this.defaults.characterCountAttribute
```

Comment:
It looks like we’re doing this on characters - why are we supporting words?

Reply:
It was part of the original script which allows both characters and words. I pulled this in 'as is' because this is how it will come from `govuk-frontend`. Can remove the maxwords for now.